### PR TITLE
Add documentation on enabling and configuring the prometheus exporter

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Related to #709.

## Changes

- Add markdown documentation on enabling and configuring the Prometheus exporter in `docs/`

### CODE REVIEW

File: docs/prometheus-exporter.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+The following options are available for configuring the pushing of metrics:
+
+- `--prometheus-instance-name` (`CONCORDIUM_NODE_PROMETHEUS_INSTANCE_NAME`) Set the instance label to include in metrics pushed to Pushgateway. If not present the id of the node is used.
+- `--prometheus-job-name` (`CONCORDIUM_NODE_PROMETHEUS_JOB_NAME`) Set the instance label to include in metrics pushed to Pushgateway. If not present, `concordium_node_push` is used.
+- `--prometheus-push-gateway-interval` (`CONCORDIUM_NODE_PROMETHEUS_PUSH_GATEWAY_INTERVAL`) Duration in seconds between pushes to Pushgateway. If not present `2` is used.
+- `--prometheus-push-gateway-password` (`CONCORDIUM_NODE_PROMETHEUS_PUSH_GATEWAY_PASSWORD`) Password to use for push gateway, both username or password must be provided needed to enable authentication.
+- `--prometheus-push-gateway-username` (`CONCORDIUM_NODE_PROMETHEUS_PUSH_GATEWAY_USERNAME`) Username to use for push gateway, both username or password must be provided needed to enable authentication.
```
Comment (abizjak) [inline docs/prometheus-exporter.md:23]: "provided needed" seems wrong

**Review by abizjak — APPROVED** [review]

File: docs/prometheus-exporter.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Enable the exporter
+
+Concordium node does not run the Prometheus exporter by default. To enable it at node startup, the listening port for the exporter must be provided either through a command line argument `--prometheus-listen-port` or the environment variable `CONCORDIUM_NODE_PROMETHEUS_LISTEN_PORT`.
+
+Likewise, the IP address to listen on can be specified using command line argument `--prometheus-listen-addr` or environment variable `CONCORDIUM_NODE_PROMETHEUS_LISTEN_ADDRESS`. Defaults to `127.0.0.1` when not provided.
+
+To verify the exporter is running open the provided listen address and port in a browser. Which should display the text "Operational".
+
+## Push metrics to a Pushgateway
```
Comment (abizjak) [inline docs/prometheus-exporter.md:13]: We should probably mention also that you can't do both at the same time, push and serve. If I remember correctly.

**Review by abizjak — APPROVED** [review]

File: docs/prometheus-exporter.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+This document describes how to configure `concordium-node` to run a [Prometheus exporter](https://prometheus.io/) and documentation for what each of the exported metrics mean.
+
+## Enable the exporter
+
+Concordium node does not run the Prometheus exporter by default. To enable it at node startup, the listening port for the exporter must be provided either through a command line argument `--prometheus-listen-port` or the environment variable `CONCORDIUM_NODE_PROMETHEUS_LISTEN_PORT`.
+
+Likewise, the IP address to listen on can be specified using command line argument `--prometheus-listen-addr` or environment variable `CONCORDIUM_NODE_PROMETHEUS_LISTEN_ADDRESS`. Defaults to `127.0.0.1` when not provided.
+
+To verify the exporter is running open the provided listen address and port in a browser. Which should display the text "Operational".
```
Comment (EmilLa1) [inline docs/prometheus-exporter.md:11]:
```suggestion
To verify whether the exporter is running, then open the provided listen address and port in a browser. Which should display the text "Operational".
```

**Review by EmilLa1 — APPROVED** [review]
Looks good. 
Some punctuation got me thrown off a bit .

File: docs/prometheus-exporter.md (hunk 2 above)
Comment (limemloh) [reply docs/prometheus-exporter.md:13]: I don't see that limitation in our own code or documented in the prometheus library

Comment (abizjak) [reply docs/prometheus-exporter.md:13]: You're right, it's just the logging that is broken.
```rust
    if conf.prometheus.prometheus_listen_port.is_some() {
        info!("Enabling prometheus server");
    } else if let Some(ref push_gateway) = conf.prometheus.prometheus_push_gateway {
        info!("Enabling prometheus push gateway at {}", push_gateway);
    };
```

Comment (limemloh) [reply docs/prometheus-exporter.md:13]: I can make a PR fixing that

Comment (abizjak) [reply docs/prometheus-exporter.md:13]: Please do.

**Pull request merged**